### PR TITLE
Detect all types of dependencies

### PR DESCRIPTION
**Pull request by rpotter12**
Signed-off-by: Rohit Potter rohitpotter12@gmail.com

- handle `.podspec` files too.
- detect all types of dependencies in `gemspec` and `podspec` files.

### CODE REVIEW

**Comment by rpotter12** [comment]
@balasankarc @pombredanne Please check these changes. Are they look good??

File: gemfileparser/__init__.py [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# .
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# .
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (pombredanne) [inline gemfileparser/__init__.py]: Why do you add this header which is furthermore misleading since the library is now under a choice of MIT or GPL?

File: gemfileparser/__init__.py [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
+
+"""
+Python library to parse Ruby's Gemfiles and gemspec files.
+"""
```
Comment (pombredanne) [inline gemfileparser/__init__.py]: Please leave the formatting with black to a separate PR entirely to avoid any confusion

File: setup.py [hunk 3]
```diff
@@ -83,7 +83,7 @@
 .. _GNU GPL version 3 (or above) License: http://www.gnu.org/licenses/gpl
 ''',
     'install_requires': [],
-    'tests_require': ['nose'],
+    'tests_require': [],
```
Comment (pombredanne) [inline setup.py]: Can you make that a separate PR?

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks!
Please:
1. add a DCO signoff to your commits
2. do not mix formatting and quote changes with other changes: use separate PRs

File: setup.py (hunk 3 above)
Comment (rpotter12) [reply setup.py]: Sure I will make a new PR for this :)

File: gemfileparser/__init__.py (hunk 2 above)
Comment (rpotter12) [reply gemfileparser/__init__.py]: Sure :)

**Comment by balasankarc** [comment]
@rpotter12 I think you might want to rebase against latest master, since I had synced GitLab and GitHub so that there might be some new code in here that was added after you forked.

**Comment by rpotter12** [comment]
> @rpotter12 I think you might want to rebase against latest master, since I had synced GitLab and GitHub so that there might be some new code in here that was added after you forked.

@balasankarc I don't think I have to rebase my fork for now as you can see in `Files Changed` that none of the previous code is changed. I have only added some new code to detect all types of data. 

Please review this again :)

File: gemfileparser/__init__.py [hunk 4]
```diff
@@ -47,9 +47,11 @@ def __init__(self):
     group_block_regex = re.compile(
         r"group[ ]?:[ ]?(?P<groupblock>.*?) do")
     add_dvtdep_regex = re.compile(
-        r".*add_development_dependency (?P<line>.*)")
+        r".*add_development_dependency(?P<line>.*)")
     add_rundep_regex = re.compile(
-        r".*add_runtime_dependency (?P<line>.*)")
+        r".*add_runtime_dependency(?P<line>.*)")
```
Comment (pombredanne) [inline gemfileparser/__init__.py:52]: I know it is tempting to fix formatting here, but keep that for a separate PR that would handle that as well as using black throughout

File: gemfileparser/__init__.py [hunk 5]
```diff
@@ -58,12 +60,14 @@ def __init__(self, filepath, appname=''):
...
             'runtime': [],
+            'dependency': [],
             'test': [],
             'production': [],
             'metrics': []
         }
         self.contents = self.gemfile.readlines()
-        if filepath.endswith('gemspec'):
+        path = ('gemspec', 'podspec')
+        if filepath.endswith(path):
```
Comment (pombredanne) [inline gemfileparser/__init__.py]: `path` is IMHO not a happy variable name as it could shadow stdlib imports .. and you further never reuse that afterwards... 
So I suggest either:
1. to use a better name such as `spec_extensions` (note the plural) as in :
` path = ('gemspec', 'podspec', )` (note also the trailing comma that is what makes this a tuple and is more explicit.)
2. or just use directly this `if filepath.endswith(('gemspec', 'podspec',)):`

Finally since we really care bout extensions, what about adding a leading period as in:
`if filepath.endswith(('.gemspec', '.podspec',)):`

**Review by pombredanne — CHANGES_REQUESTED** [review]
See my final and tiny nitpickings! we are almost there

File: gemfileparser/__init__.py (hunk 5 above)
Comment (rpotter12) [reply gemfileparser/__init__.py]: Okay :+1:
I will make changes in few minutes :)

**Review by pombredanne — APPROVED** [review]
Thank you. LGTM. Merging
